src/firefly/java/edu/caltech/ipac/firefly/server/query/ObsCorePackager.java

```diff
@@ -82,7 +88,7 @@ private List<FileGroup> computeFileGroup(DownloadRequest request) throws DataAcc
                 if (access_format != null) {
                     if (access_format.contains("datalink")) {
                         ext_file_name = getUniqueFileNameForGroup(ext_file_name, dataLinkFolders);
-                        List<FileInfo> tmpFileInfos = parseDatalink(url, ext_file_name, products, useSourceUrlFileName);
+                        List<FileInfo> tmpFileInfos = parseDatalink(url, ext_file_name, products, useSourceUrlFileName, ra, dec, cutoutValue);
```

Comment:
In my opinion, most of these parameters are specific to `parseDatalink`.  Extracting them from the top just to pass it along here makes the code confusing.
I would have place all of `parseDatalink` logic in that one function, like this:
```
List<FileInfo> tmpFileInfos = parseDatalink(url, request, dgDataUrl.getRow(int idx));
```
When you do this, you will see that almost half of the code in this function is no longer needed.